src/docs/arch-storage-extension.dox

```diff
@@ -0,0 +1,104 @@
+/*! @arch_page arch-cloud-storage-extension Storage Source
+
+@section extension_storage_source Cloud Storage Source Implementation
+
+The WiredTiger extension allows the control of how files are stored via the storage source
+abstraction layer. More information can be found through @ref custom_storage_sources. There are
+three storage sources that extends the WiredTiger's capability to store the data files onto the
+cloud storage through Azure, Google and AWS providers. The cloud storage extensions provides a
+means to construct, configure and control a custom file system linked to a cloud bucket. All three
```

Comment:
Grammar: "The cloud storage extensions provides a means to construct" -> "The cloud storage extensions provide a means to construct"

Reply:
dine